routes/user.js

```diff
@@ -0,0 +1,71 @@
+var moduleUser = require('../models/user');
+var express = require('express');
+var router = express.Router();
+var promise = require('bluebird');
+var auth = require('../config/auth');
+var mongoose = require('mongoose');
+
+mongoose.Promise = promise;
+
+router.route('/user')
+    .get(function (req, res) {
+        var username = req.user;
+        if (username) {
+            moduleUser.User
+                .find({active: true, deleted: false, username:  username})
+                .exec()
+                .then(function (result) {
+                    if (result) {
+                        res.status(200).json(result);
+                    }
+                    else
+                        res.status(400).json({
+                            error: 'Internal server error'
+                        });
+                })
+                .catch(function (err) {
+                });
+        }
+        else {
+            res.status(404).json({
+                error: 'Username not found'
+            });
+        }
+    })
+    .post(function(req, res){
+        if (req.body.email && req.body.password) {
```

Comment:
We need a check here for verifying if the entered email for registration is not already in our database.